modelon/impact/client/experiment_definition.py

```diff
@@ -475,16 +473,14 @@ def to_dict(self):
             ).with_modifiers({'PI.k': 40})
             simulate_ext.to_dict()
         """
-        return {
-            "extensions": [
-                {
-                    "modifiers": {"variables": self.variable_modifiers},
-                    "analysis": {
-                        "parameters": self.parameter_modifiers,
-                        "simulationOptions": self._simulation_options,
-                        "solverOptions": self._solver_options,
-                        "simulationLogLevel": self.simulation_log_level,
-                    },
-                }
-            ]
-        }
+        ext_dict = {"modifiers": {"variables": self.variable_modifiers}}
+        if (
```

Comment:
@efredriksson-modelon , I have made a not so nice fix over here to hide the "analysis" stuff. The reason being simulation_log_level needs to have a default specified. Please let me know if there is another way around this! Thanks!

Reply:
Default argument for simulation_log_level should be None, which would signify that it is not specified? The code could look something like:

```
ext_dict = {}
if self.variable_modifiers:
  ext_dict["modifiers"] = {"variables": self.variable_modifiers}

ext_dict.setdefault("analysis", {})
if self.parameter_modifiers:
  ext_dict["analysis"]["parameters"] = self.parameter_modifiers

...
```

Reply:
Thanks! I misunderstood your comment earlier.. I thought you wanted to eliminate "analysis" key from extensions in case the "analysis" sub keys are not specified is not used. 
Also with the suggested method the output for a case with defaults and no mofifiers  would be as below :- 
```
experiment_definition = experiment_definition.with_extensions(
    [SimpleExperimentExtension()]
) 

#OUTPUT
'extensions': [{'analysis': {}}],
```
Is that how you suggest it to be?

Reply:
No you where correct on how it should behave. My code is just wrong. I want `ext_dict.setdefault("analysis", {})` inside each if statement. Sorry for causing more work.